Bottom: 2750c34b6d5a0aed17431e939c1784fe3693d895
Top:    8e94b7cd43014245ce7f5e9df2991544a216b5bf
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-27 20:35:07 +0100

Fix layer on/off rendering

I'm not certain this is correct at all. It works around failure
to switch layer visibility on / off after

commit 9768e060fad7bc3dfc366da76ea1db8154005018
"Fix layer visibility logic for some boards."

After which, copper layers on the component / solder group
would still be drawn unless the silk layer was also turned off.

This appears to be due to the GL HID's custom drawing routines
not checking the return of the (HID)->set_layer function in certain
instances.

This fix is almost certainly WRONG.


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index ba0fe84..81cfcc7 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1600,7 +1600,7 @@ DrawLayerGroup (int group, const BoxType * screen)
         strcmp (Layer->Name, "route") == 0)
       rv = 0;
 
-    if (layernum < max_copper_layer /*&& Layer->On*/) {
+    if (layernum < max_copper_layer && Layer->On) {
 
       if (!first_run)
         gui->set_layer (0, group, 0);
